mentat/conversation.py

```diff
@@ -149,20 +136,26 @@ async def _stream_model_response(
         time_elapsed = default_timer() - start_time
         return (message, file_edits, time_elapsed)
 
-    async def get_model_response(
-        self, parser: Parser, config: ConfigManager
-    ) -> list[FileEdit]:
+    async def get_model_response(self) -> list[FileEdit]:
+        stream = SESSION_STREAM.get()
+        code_context = CODE_CONTEXT.get()
+        cost_tracker = COST_TRACKER.get()
+        code_file_manager = CODE_FILE_MANAGER.get()
+        parser = PARSER.get()
```

Comment:
It's pretty common we need to `.get` multiple object instances stored in Contextvars in a lot of methods/functions, so I'm wondering if it would be better to have a single Contextvar (e.g. a `SessionContext`) that's just a dict holding the instances for `SessionStream`, `CostTracker`, `Parser`, etc.). Thoughts?

It's kind of like having a single state object in React you just pass around everywhere instead of having many state variables.

Reply:
All of these instances are tied to a specific session anyways

Reply:
Ya that makes sense, lots of things tied to the session. I was also thinking this reminded me somehow of react.

Reply:
Hmm, I wouldn't be opposed to it; I like the idea of a SessionContext class, but on the other hand I feel like it bundles up a lot of unrelated stuff for not much reason. How about we keep it like this for now (since it's just a couple of extra lines of code), and if we want to in the future we can move to it?

Reply:
I like the `SessionContext` idea, it'd make it explicit which things are part of the state

Reply:
@PCSwingle which unrelated stuff would be bundled up for no reason?

Reply:
By having a bunch of different context vars we're passing around it feels like we'd just me moving the issue of passing a bunch of object instances around to having to make a bunch of `SOME_CONTEXT_VAR.get()` calls in every function.

Reply:
In most cases functions won't need to access everything in the context, but I think it's a fine tradeoff and makes the codebase simpler.

Reply:
Ok, I'll add in the SessionContext; although I can't say I'm 100% convinced haha